zerver/tornado/event_queue.py

```diff
@@ -949,14 +946,15 @@ def get_client_payload(apply_markdown: bool, client_gravatar: bool) -> Dict[str,
         # If the recipient was offline and the message was a single or group PM to them
         # or they were @-notified potentially notify more immediately
         private_message = message_type == "private" and user_profile_id != sender_id
-        mentioned = "mentioned" in flags
+        mentioned = user_data.get("mentioned", False)
         stream_push_notify = user_data.get("stream_push_notify", False)
```

Comment:
Ideally, this should be something like `user_data.get("mentioned", "mentioned" in flags)` for backwards compatibility, but I'm not sure if that's worth doing because:
- It defeats the purpose of deduplication this logic
- Tornado won't crash if we don't do that -- the worst that will happen is someone not receiving a notification during upgrade